keps/sig-node/5027-dra-admin-controlled-device-attributes/README.md

```diff
@@ -0,0 +1,848 @@
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+With Dynamic Resource Allocation (DRA), DRA drivers publish information about
+the devices that they manage in ResourceSlices. This information is used by the
+scheduler when selecting devices for user requests in ResourceClaims.
+
+This KEP adds a Kubernetes API that privileged users, typically cluster
+administrators, can use to override or extend that information. This can be
+permanent as part of the installation of a DRA driver to adapt the driver to
+the cluster or temporary as part of cluster maintenance.
+
+This generic mechanism can be used to mark devices as offline, regardless
+whether it is because of device health as observed by some component other than
+the driver itself or because the cluster administrator is doing some manual
+maintenance work.
+
+The other usage is to influence which devices are picked when there are
+multiple viable alternatives. This is a first step towards providing a more
+comprehensive [scoring](https://github.com/kubernetes/enhancements/issues/4970)
+solution.
+
+For both usages this KEP standardizes device attributes that need to be checked
+by the scheduler before using a device for a ResourceClaim.
+
+## Motivation
+
+### Goals
+
+- Enable [admin-controlled](../5055-dra-device-taints-and-tolerations) device taints.
+
+- Enable updating how devices are seen in the cluster without having to use
+  driver-specific APIs which influence what a driver puts into ResourceSlices.
+
+### Non-Goals
+
+- At least for alpha: extend `kubectl` to provide a unified view of devices
+  together with all overrides that apply to them.
```

Comment:
@eero-t in https://github.com/kubernetes/enhancements/pull/5034#pullrequestreview-2562581352
> There was earlier discussion of common (driver independent) tool(ing) for listing, adding and removing device taints. Would it make sense to mention something about that in the tainting KEP?

I added this here for attributes and in the other KEP for taints.

Reply:
See https://github.com/kubernetes/enhancements/pull/5034/files#r1923530521 for more details.